syntax = "proto3";

package pixell.agent;

// Empty message for simple requests
message Empty {}

// Health status response
message HealthStatus {
  bool ok = 1;
  string message = 2;
  int64 timestamp = 3;
}

// Agent capabilities
message Capabilities {
  repeated string methods = 1;
  map<string, string> metadata = 2;
}

// Action request
message ActionRequest {
  string action = 1;
  map<string, string> parameters = 2;
  string request_id = 3;
}

// Action result
message ActionResult {
  bool success = 1;
  string result = 2;
  string error = 3;
  string request_id = 4;
  int64 duration_ms = 5;
}

// Ping response
message Pong {
  string message = 1;
  int64 timestamp = 2;
}

// A2A service definition
service AgentService {
  // Health check
  rpc Health(Empty) returns (HealthStatus);
  
  // Describe agent capabilities
  rpc DescribeCapabilities(Empty) returns (Capabilities);
  
  // Invoke an action
  rpc Invoke(ActionRequest) returns (ActionResult);
  
  // Simple ping
  rpc Ping(Empty) returns (Pong);
}
